sql/core/src/main/scala/org/apache/spark/sql/execution/command/CommandUtils.scala

```diff
@@ -43,13 +43,17 @@ object CommandUtils extends Logging {
   /** Change statistics after changing data by commands. */
   def updateTableStats(sparkSession: SparkSession, table: CatalogTable): Unit = {
     val catalog = sparkSession.sessionState.catalog
-    if (sparkSession.sessionState.conf.autoSizeUpdateEnabled) {
-      val newTable = catalog.getTableMetadata(table.identifier)
-      val newSize = CommandUtils.calculateTotalSize(sparkSession, newTable)
-      val newStats = CatalogStatistics(sizeInBytes = newSize)
-      catalog.alterTableStats(table.identifier, Some(newStats))
-    } else if (table.stats.nonEmpty) {
-      catalog.alterTableStats(table.identifier, None)
+    val statsMaybeUpdatedByHive = catalog.getTableMetadata(table.identifier)
```

Comment:
This module is `sql/core`. We should not have an assumption with `Hive`. Please rename `statsMaybeUpdatedByHive` to more neutral one.

Reply:
+1 , I think it's nothing about hive. It's just, we shouldn't call `alterTableStats` if the stats is not changed.

Reply:
How about:
```scala
  def updateTableStats(sparkSession: SparkSession, table: CatalogTable): Unit = {
    val catalog = sparkSession.sessionState.catalog
    val statsMaybeUpdatedByExternalCatalog = catalog.getTableMetadata(table.identifier)
    if (statsMaybeUpdatedByExternalCatalog.stats == table.stats) {
      if (sparkSession.sessionState.conf.autoSizeUpdateEnabled) {
        val newSize =
          CommandUtils.calculateTotalSize(sparkSession, statsMaybeUpdatedByExternalCatalog)
        val newStats = CatalogStatistics(sizeInBytes = newSize)
        catalog.alterTableStats(table.identifier, Some(newStats))
      } else if (table.stats.nonEmpty) {
        catalog.alterTableStats(table.identifier, None)
      }
    } else {
      logInfo(
        s"Skip to update statistics for ${table.qualifiedName} as it updated by external catalog.")
    }
  }
```